anywhere-fitness-front-end/src/Components/BBBFrontPage/SignUpForm.js

```diff
@@ -107,42 +106,42 @@ const SignUpForm = () => {
 
   // Disable button if invalid inputs
   // useEffect(() => {
-  //   if ((formState.firstName.length < 3) && (formState.lastName.length < 3) && (formState.email.length < 3) && (formState.terms === false)) {
+  //   if ((formState.name.length < 3) && (formState.lastName.length < 3) && (formState.email.length < 3) && (formState.terms === false)) {
   //     setButtonDisabled(true);
   //   } else {
   //     setButtonDisabled(false);
   //   }
   // }, [formState]);
 
   useEffect(() => {
-    if (formState.firstName.length > 2) {
+    if (formState.name.length > 2) {
       setButtonDisabled(false);
     } else {
       setButtonDisabled(true);
     }
   }, [formState]);
 
   useEffect(() => {
-    if (formState.lastName.length > 2) {
+    if (formState.email.length > 2) {
       setButtonDisabled(false);
     } else {
       setButtonDisabled(true);
     }
   }, [formState]);
 
   useEffect(() => {
-    if (formState.email.length > 2) {
+    if (formState.username.length > 2) {
       setButtonDisabled(false);
     } else {
       setButtonDisabled(true);
     }
   }, [formState]);
 
   useEffect(() => {
-    if (formState.terms) {
-      setButtonDisabled(!true);
+    if (formState.password.length > 2) {
```

Comment:
for the sake of dryness you could combine these into one useEffect

Reply:
Thanks, good catch. I actually tried to use an && statement to hold them all together but it wasn't working. I'll try it again once I get Brandon's work connected but If I don't figure it out today ill definitely reach out.